sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -2967,6 +2974,92 @@ def get_page_ranges( # type: ignore
             process_storage_error(error)
         return get_page_ranges_result(ranges)
 
+    @distributed_trace
+    def list_page_ranges(
+            self,
+            *,
+            offset: Optional[int] = None,
+            length: Optional[int] = None,
+            previous_snapshot: Optional[Union[str, Dict[str, Any]]] = None,
```

Comment:
What's the `Dict[str, Any]` supported here?

Reply:
I believe this is to handle the output of `create_snapshot()` directly. That API outputs a `Dict[str, Any]` with some of the response headers with one of the being `"snapshot"`. We have this to allow that output to be passed directly and we parse this in `_get_page_ranges_options()`. A bit weird but I left it to be consistent with the existing API.

```python
if previous_snapshot_diff:
          try:
              options['prevsnapshot'] = previous_snapshot_diff.snapshot # type: ignore
          except AttributeError:
              try:
                  options['prevsnapshot'] = previous_snapshot_diff['snapshot'] # type: ignore
              except TypeError:
                  options['prevsnapshot'] = previous_snapshot_diff
```